tests/ignite/contrib/metrics/test_average_precision.py

```diff
@@ -240,24 +241,25 @@ def _test(y_preds, y_true, n_epochs, metric_device, update_fn):
         assert pytest.approx(res) == true_res
 
     def get_tests(is_N):
+        torch.manual_seed(12 + rank)
         if is_N:
-            y_true = torch.randint(0, n_classes, size=(offset * idist.get_world_size(),)).to(device)
-            y_preds = torch.rand(offset * idist.get_world_size()).to(device)
+            y_true = torch.randint(0, n_classes, size=(n_iters * batch_size,)).to(device)
+            y_preds = torch.rand(n_iters * batch_size).to(device)
 
             def update_fn(engine, i):
                 return (
-                    y_preds[i * s + rank * offset : (i + 1) * s + rank * offset],
-                    y_true[i * s + rank * offset : (i + 1) * s + rank * offset],
+                    y_preds[i * batch_size : (i + 1) * batch_size],
+                    y_true[i * batch_size : (i + 1) * batch_size],
                 )
 
         else:
-            y_true = torch.randint(0, n_classes, size=(offset * idist.get_world_size(), 10)).to(device)
-            y_preds = torch.randint(0, n_classes, size=(offset * idist.get_world_size(), 10)).to(device)
+            y_true = torch.randint(0, n_classes, size=(n_iters * batch_size, 10)).to(device)
+            y_preds = torch.randint(0, n_classes, size=(n_iters * batch_size, 10)).to(device)
 
             def update_fn(engine, i):
                 return (
-                    y_preds[i * s + rank * offset : (i + 1) * s + rank * offset, :],
-                    y_true[i * s + rank * offset : (i + 1) * s + rank * offset, :],
+                    y_preds[i * batch_size * n_iters : (i + 1) * batch_size * n_iters, :],
+                    y_true[i * batch_size * n_iters : (i + 1) * batch_size * n_iters, :],
```

Comment:
Why `i * batch_size * n_iters` here and `i * batch_size` on Line 251 ?

Reply:
I mistakenly leave `n_iters`. I send again :)